extensions/notebook/src/jupyter/jupyterSessionManager.ts

```diff
@@ -129,6 +129,12 @@ export class JupyterSessionManager implements nb.SessionManager {
 			// no-op
 			return Promise.reject(new Error(localize('errorStartBeforeReady', "Cannot start a session, the manager is not yet initialized")));
 		}
+
+		// Prompt for Python Install to check that all dependencies are installed.
+		// This prevents the kernel from getting stuck if a user deletes a dependency after the server has been started.
+		let kernelDisplayName: string = this.specs?.kernels.find(k => k.name === options.kernelName).display_name;
+		await this._installation?.promptForPythonInstall(kernelDisplayName);
```

Comment:
What happens if a user tries to open a notebook with a kernell they don't have installed? Does this kernelspec list only include the ones that we include by default, or if showAllKernels is checked, would the behavior be different?

Reply:
1. If they already have all dependencies installed does this just no-opt then? 
2. What's the overhead of doing this check every time?

Reply:
@chlafreniere 
1. If a user tries to open a notebook with a kernel they don't have installed then they will still get the "Could not start session: Invalid response: 501 Not Implemented" error.
2. The kernelspec list actually includes all kernels, even if showAllKernels is not checked. So if a user has the .net interactive kernels installed but does not have showAllKernels checked, opening a .net interactive notebook will run fine but the kernel dropdown will show SQL, which can be confusing. Maybe we can add a check here and recommend the user to check the showAllKernels setting?

@Charles-Gagnon 
1. Yes
2. There is not much overhead since we actually cache the kernel setups so that we only have to check all dependencies are installed for each kernel when the server is started. When dependent packages are removed, the cache gets updated.